genai_processors/contrib/openrouter_model.py

```diff
@@ -0,0 +1,424 @@
+                      'url': f'data:{part.mimetype};base64,{encoded_image}'
+                  },
+              }
+          ],
+      }
+
+  # Fallback to text representation
+  return {
+      'role': role,
+      'content': str(part.part),
+  }
+
+
+def _parse_sse_line(line: str) -> dict[str, Any] | None:
+  """Parse a Server-Sent Events line."""
+  line = line.strip()
+
+  # Skip comments and empty lines
+  if not line or line.startswith(':'):
+    return None
+
+  # Parse data lines
+  if line.startswith('data: '):
+    data = line[6:]
+    if data == '[DONE]':
+      return {'type': 'done'}
+
+    try:
+      return json.loads(data)
+    except json.JSONDecodeError:
+      return None
+
+  return None
+
+
+class OpenRouterModel(processor.Processor):
+  """`Processor` that calls OpenRouter API with streaming support.
+
+  OpenRouter provides access to hundreds of AI models through a unified API,
+  including models from OpenAI, Anthropic, Google, Meta, and many others.
+  """
+
+  def __init__(
+      self,
+      *,
+      api_key: str,
+      model_name: str,
+      base_url: str | None = None,
+      site_url: str | None = None,
+      site_name: str | None = None,
+      generate_content_config: GenerateContentConfig | None = None,
+  ):
+    """Initialize the OpenRouter model.
+
+    Args:
+      api_key: Your OpenRouter API key.
+      model_name: Model to use (e.g., "openai/gpt-4o", "anthropic/claude-3-5-sonnet").
+      base_url: OpenRouter API base URL (defaults to https://openrouter.ai/api/v1).
+      site_url: Your site URL (optional, for OpenRouter rankings).
+      site_name: Your site name (optional, for OpenRouter rankings).
+      generate_content_config: Model configuration parameters.
+
+    Returns:
+      A `Processor` that calls the OpenRouter API with streaming support.
+    """
+    self._api_key = api_key
+    self._model_name = model_name
+    self._base_url = base_url or _DEFAULT_BASE_URL
+    self._site_url = site_url
+    self._site_name = site_name
+    self._config = generate_content_config or {}
+
+    # Build headers
+    headers = {
+        'Authorization': f'Bearer {self._api_key}',
+        'Content-Type': 'application/json',
+        'User-Agent': 'genai-processors',
+    }
+
+    if self._site_url:
+      headers['HTTP-Referer'] = self._site_url
+    if self._site_name:
+      headers['X-Title'] = self._site_name
+
+    self._client = httpx.AsyncClient(
+        base_url=self._base_url,
+        headers=headers,
+        timeout=_DEFAULT_TIMEOUT,
+    )
+
+  @property
+  def key_prefix(self) -> str:
+    """Key prefix for caching."""
+    return f'OpenRouterModel_{self._model_name}'
+
+  async def call(
+      self, content: AsyncIterable[content_api.ProcessorPartTypes]
+  ) -> AsyncIterable[content_api.ProcessorPartTypes]:
+    """Process content through OpenRouter API."""
+    messages = []
+    async for part in content:
+      if isinstance(part, content_api.ProcessorPart):
```

Comment:
You don't need that. You can safely change type annotation of `call` method to accept only `content_api.ProcessorPart`. The library laready has built-in conversion. When you use processors you can all the things convertible to ProcessorParts e.g. strings or PIL images. But when you write the processor itself, you get homogenous `ProcessorPart` objects. So implementation of your `call` method can be much simpler.